pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -307,6 +306,32 @@ private void reloadSegment(String tableNameWithType, SegmentMetadata segmentMeta
     }
   }
 
+  /**
+   * Try to reload a segment without a local index directory. The segment can be a consuming
+   * segment from a REALTIME table, or an immutable segment on remote tier backend.
+   * @return true if the segment is loaded.
+   */
+  @VisibleForTesting
+  boolean reloadSegmentWithNullIndexDir(String tableNameWithType, String segmentName,
+      SegmentDataManager segmentDataManager, @Nullable Schema schema) {
+    IndexSegment segment = segmentDataManager.getSegment();
+    if (segment instanceof ImmutableSegment) {
+      LOGGER.info("Found an immutable segment: {} in table: {} on remote tier backend", segmentName,
```

Comment:
```suggestion
      LOGGER.info("Found an immutable segment: {} in table: {}.", segmentName,
```

Reply:
👌

Reply:
I'll save the period in the end, as the convention of log msg in this class seems like not ending with period.